helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -268,30 +301,92 @@ public StoppableCheck getInstanceStoppableCheck(String clusterId, String instanc
 
   public Map<String, StoppableCheck> batchGetInstancesStoppableChecks(String clusterId,
       List<String> instances, String jsonContent) throws IOException {
-    Map<String, StoppableCheck> finalStoppableChecks = new HashMap<>();
     // helix instance check
+    Map<String, StoppableCheck> finalStoppableChecks = new HashMap<>();
+
     List<String> instancesForCustomInstanceLevelChecks =
         batchHelixInstanceStoppableCheck(clusterId, instances, finalStoppableChecks);
-    // custom check
+    // custom check, includes partition check.
     batchCustomInstanceStoppableCheck(clusterId, instancesForCustomInstanceLevelChecks,
         finalStoppableChecks, getMapFromJsonPayload(jsonContent));
     return finalStoppableChecks;
   }
 
+  private MaintenanceManagementInstanceInfo takeFreeSingleInstanceHelper(String clusterId,
+      String instanceName, List<String> healthChecks, Map<String, String> healthCheckConfig,
+      List<String> operations, Map<String, String> operationConfig, boolean performOperation,
+      boolean isTakeInstance) {
+
+    if (operations == null) {
+      operations = new ArrayList<>();
+    }
+
+    try {
+      MaintenanceManagementInstanceInfo instanceInfo;
+      if (isTakeInstance) {
+        if (healthChecks == null) {
+          healthChecks = new ArrayList<>();
+        }
+        instanceInfo =
+            batchInstanceHealthCheck(clusterId, ImmutableList.of(instanceName), healthChecks,
+                healthCheckConfig).getOrDefault(instanceName, new MaintenanceManagementInstanceInfo(
+                MaintenanceManagementInstanceInfo.OperationalStatus.SUCCESS));
+        if (!instanceInfo.isSuccessful()) {
```

Comment:
I remember we should be able to let user define which are the ones can be skipped even if failed.

Reply:
I think bypassing an operation check should be handled at implementation level not API level. The operation check will return `SUCCESS` if bypassed and add the failure note in message. 
Or we can add a new state 'ByPassed or non-blocking check' in instanceInfo?

Reply:
Had an offline discussion. 
Conclusion: In health check config, allow user to specify the health check name to be non blocking. For each operation class, add a field in operation config indicating if this operation check is non-blocking or not.